File: use-cases/easy-ship/code/java/src/main/java/lambda/SubmitFeedRequestHandler.java
package lambda;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.google.gson.Gson;
import io.swagger.client.api.FeedsApi;
import io.swagger.client.model.feeds.CreateFeedDocumentResponse;
import io.swagger.client.model.feeds.CreateFeedDocumentSpecification;
import io.swagger.client.model.feeds.CreateFeedResponse;
import io.swagger.client.model.feeds.CreateFeedSpecification;
import io.swagger.client.model.feeds.FeedOptions;
import lambda.utils.ApiUtils;
import lambda.utils.HttpFileTransferUtil;
import lambda.utils.StateMachineInput;
import lambda.utils.XmlUtil;
import java.nio.charset.StandardCharsets;
import java.util.Collections;

import static lambda.utils.Constants.*;


public class SubmitFeedRequestHandler implements RequestHandler<StateMachineInput, StateMachineInput> {

    public StateMachineInput handleRequest(StateMachineInput input, Context context) {
        LambdaLogger logger = context.getLogger();
        logger.log("SubmitFeedRequest Lambda input: " + new Gson().toJson(input));

        try {
            // Create Feeds document
            FeedsApi feedsApi = ApiUtils.getFeedsApi(input);
            String contentType = String.format("text/xml; charset=%s", StandardCharsets.UTF_8);
            CreateFeedDocumentSpecification createFeedDocumentbody = new CreateFeedDocumentSpecification().contentType(contentType);
            CreateFeedDocumentResponse createFeedDocumentresponse = feedsApi.createFeedDocument(createFeedDocumentbody);
            logger.log("Feed API -  Create Feeds document response: " + new Gson().toJson(createFeedDocumentresponse));
            // Upload Feeds Document
            String url = createFeedDocumentresponse.getUrl();
            String content = XmlUtil.generateEasyShipAmazonEnvelope(
                    input.getSellerId(),input.getAmazonOrderId(), FEED_OPTIONS_DOCUMENT_TYPE_VALUE);
            HttpFileTransferUtil.upload(content.getBytes(StandardCharsets.UTF_8), url);
            // Create Feeds
            FeedOptions feedOptions = new FeedOptions();
            feedOptions.put(FEED_OPTIONS_KEY_AMAZON_ORDER_ID, input.getAmazonOrderId());
            feedOptions.put(FEED_OPTIONS_KEY_DOCUMENT_TYPE, FEED_OPTIONS_DOCUMENT_TYPE_VALUE);
            String feedDocumentId = createFeedDocumentresponse.getFeedDocumentId();
            CreateFeedSpecification createFeedbody = new CreateFeedSpecification()
                    .feedType(POST_EASYSHIP_DOCUMENTS)
                    .marketplaceIds(Collections.singletonList(input.getMarketplaceId()))
                    .feedOptions(feedOptions)
                    .inputFeedDocumentId(feedDocumentId);
            logger.log("Feed API -  Create Feeds  request body: " + new Gson().toJson(createFeedbody));
            CreateFeedResponse createFeedResponse = feedsApi.createFeed(createFeedbody);
            logger.log("Feed API -  Create Feeds  response: " + new Gson().toJson(createFeedDocumentresponse));
            input.setFeedId(createFeedResponse.getFeedId());
            return input;
        } catch (Exception e) {
            throw new InternalError("Submit Feed Request failed", e);
        }
    }
}

File: use-cases/easy-ship/code/php/lambda/SubmitFeedRequestHandler.php
<?php

namespace Lambda;

use GuzzleHttp\Exception\GuzzleException;
use GuzzleHttp\Exception\RequestException;
use Lambda\Utils\ApiUtils;
use Lambda\Utils\HttpFileTransferUtil;
use Lambda\utils\LambdaContext;
use Lambda\Utils\model\StateMachineInput;
use Lambda\Utils\StateMachineInputConverter;
use Lambda\Utils\XmlUtil;
use Lambda\utils\Constants;
use SpApi\Model\feeds\v2021_06_30\CreateFeedDocumentSpecification;
use SpApi\Model\feeds\v2021_06_30\CreateFeedSpecification;

class SubmitFeedRequestHandler
{
    public function handleRequest(array $event, LambdaContext $context): StateMachineInput
    {
        $input = StateMachineInputConverter::convertFromArray($event);

        $logger = $context->getLogger();
        $logger->info('SubmitFeedRequest Lambda input: ' . json_encode($event, JSON_PRETTY_PRINT));

        try {
            // Initialize Feeds API client
            $feedsApi = ApiUtils::getFeedsApi($input);

            // Create Feed Document
            $contentType = 'text/xml; charset=UTF-8';
            $createFeedDocumentSpec = new CreateFeedDocumentSpecification(['content_type' => $contentType]);

            $createFeedDocumentResponse = $feedsApi->createFeedDocument($createFeedDocumentSpec);
            $logger->info('Feed API - Create Feeds document response: ' . json_encode($createFeedDocumentResponse));

            // Upload Feed Document
            $url = $createFeedDocumentResponse->getUrl();
            $content = XmlUtil::generateEasyShipAmazonEnvelope(
                $input->getSellerId(),
                $input->getAmazonOrderId(),
                Constants::FEED_OPTIONS_DOCUMENT_TYPE_VALUE
            );

            HttpFileTransferUtil::upload($content, $url);

            // Create Feed
            $createFeedSpec = (new CreateFeedSpecification())
                ->setFeedType(Constants::POST_EASYSHIP_DOCUMENTS)
                ->setMarketplaceIds([$input->getMarketplaceId()])
                ->setFeedOptions([
                    Constants::FEED_OPTIONS_KEY_AMAZON_ORDER_ID => $input->getAmazonOrderId(),
                    Constants::FEED_OPTIONS_KEY_DOCUMENT_TYPE => Constants::FEED_OPTIONS_DOCUMENT_TYPE_VALUE
                ])
                ->setInputFeedDocumentId($createFeedDocumentResponse->getFeedDocumentId());


            $logger->info('Feed API - Create Feeds request body: ' . json_encode($createFeedSpec));
            $createFeedResponse = $feedsApi->createFeed($createFeedSpec);
            $logger->info('Feed API - Create Feeds response: ' . json_encode($createFeedResponse));

            // Set Feed ID in input
            $input->setFeedId($createFeedResponse->getFeedId());

            return $input;
        } catch (RequestException $e) {
            $logger->error('RequestException: ' . $e->getMessage());
            throw new \RuntimeException('Submit Feed Request failed', 0, $e);
        } catch (\Exception $e) {
            $logger->error('Exception: ' . $e->getMessage());
            throw new \RuntimeException('An unexpected error occurred', 0, $e);
        } catch (GuzzleException $e) {
        }
    }
}
